targets/TARGET_Cypress/TARGET_PSOC6_FUTURE/sleep_api.c

```diff
@@ -0,0 +1,42 @@
+/*
+ * mbed Microcontroller Library
+ * Copyright (c) 2017-2018 Future Electronics
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "cmsis.h"
+#include "device.h"
+#include "cy_syspm.h"
+
+#if DEVICE_SLEEP
+
+void hal_sleep(void)
+{
+    Cy_SysPm_Sleep(CY_SYSPM_WAIT_FOR_INTERRUPT);
+}
+
+void hal_deepsleep(void)
+{
+#if DEVICE_LPTICKER
+    if(CY_SYSPM_SUCCESS == Cy_SysPm_DeepSleep(CY_SYSPM_WAIT_FOR_INTERRUPT)) {
```

Comment:
this file is just moved in this PR but why deepsleep only if lpticker is enabled? is there MCU that does not have lp ticker support thus no deepsleep?

Reply:
@0xc0170: all PSoC6 MCUs have LPTICKER, it is enabled in targets.json for MCU_PSOC6 base target.
The conditional check can be easily removed, without functional impact. For future releases, Cypress plans PSoC6 HAL cleanup.